packages/core/src/typeschema/crawler.ts

```diff
@@ -103,27 +112,78 @@ class ResourceCrawler {
 }
 
 export function getNestedProperty(
-  value: TypedValue,
+  value: TypedValueWithIndexedPath | undefined,
+  key: string,
+  options: { profileUrl?: string; withIndexedPath: true }
+): (TypedValueWithIndexedPath | TypedValueWithIndexedPath[])[];
+export function getNestedProperty(
+  value: TypedValue | undefined,
+  key: string,
+  options?: { profileUrl?: string; withIndexedPath?: false }
+): (TypedValue | TypedValue[] | undefined)[];
+export function getNestedProperty(
+  value: TypedValue | undefined,
   key: string,
-  options?: { profileUrl?: string }
+  options?: { profileUrl?: string; withIndexedPath?: boolean }
 ): (TypedValue | TypedValue[] | undefined)[] {
+  if (value === undefined) {
+    return [undefined];
+  }
+
   if (key === '$this') {
     return [value];
   }
+
+  const propertyGetter = options?.withIndexedPath ? getTypedPropertyValueWithIndexedPath : getTypedPropertyValue;
```

Comment:
Similar question here.... what are the conditions we would use `getTypedPropertyValueWithIndexedPath` vs `getTypedPropertyValue`?

Would there be any harm if we simply always returned the path?  (aside from perf hit)

Reply:
On the surface, no, I don't think there'd be an issue with always returning a path besides `TypedValue` being a bit of a misnomer since the object includes more than just `type` and `value`.

Another concern with making that blanket change is the behavior around `isEmpty`/`undefined` values being represented as `{ type: 'undefined', value: undefined}` versus simply `undefined`. I think the latter would be preferable since return types of signatures could be a bit simpler: `TypedValue | TypedValue[]` instead of the current pattern of `TypedValue | TypedValue[] | undefined`. But that would be another smaller performance hit.